### ISSUE TITLE
Override Main Scene in Export.

### ISSUE THREAD

**Issue by Kazox61** [godotengine/godot-proposals#11345]
### Describe the project you are working on

I'm working on a Multiplayer Game, having the Client and the Dedicated Server in the same Project.

### Describe the problem or limitation you are having in your project

I would like to configure the run main scene for a specific Export Preset.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a Field to set a scene as Main Scene in a Export Preset. When it is used it will use this scene instead of the scene configured in the Project Settings.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I didn't find any way.

### Is there a reason why this should be core and not an add-on in the asset library?

This will help Users when working on a Mulitplayer Game.

**Comment by Calinou** [comment]
You can achieve this with custom [feature tags](https://docs.godotengine.org/en/latest/tutorials/export/feature_tags.html) already:

- Define a custom feature tag in your export preset, e.g. `demo`.
- Define the `application/run/main_scene.demo` project setting. This setting will only apply when the `demo` feature tag is present.

**Comment by Kazox61** [comment]
Perfect. That's what I need. I guess we can close this proposal then.

**Comment by Kazox61** [comment]
@Calinou sorry for tagging, but it doesn t work when exporting.
`ERROR: No loader found for resource: res://uid:/cfifco7251prs (expected type: )
   at: (core/io/resource_loader.cpp:327)`

Thats the Scene I defined in `application/run/main_scene.dedicated_server`. Is there anything else I need to consider?

**Comment by Calinou** [comment]
@Kazox61 Please open an issue on https://github.com/godotengine/godot/issues with a minimal reproduction project attached.

I suspect `application/run/main_scene` is being read using `ProjectSettings::get_setting()` instead of `ProjectSettings::get_setting_with_override()` somewhere, so it *may* need an engine fix.

**Comment by ghuser404** [comment]
@Calinou Works well in 4.4.beta3. But I think this override should still be a part of the "Application" section in the export preset, along with the "Icon", as it seems like a crucial feature that people would use often. Multiplayer (client/server), game and map editor, ability to export different levels for testing purposes, etc.

**Comment by Calinou** [comment]
> But I think this override should still be a part of the "Application" section in the export preset, along with the "Icon", as it seems like a crucial feature that people would use often.

I don't think we should provide multiple ways to do the same thing. It can already be achieved using custom features, which are not limited to a single project setting, but for any project setting out there. Custom features also allow you to perform runtime checks using `OS.has_feature()`, which you will most likely need for demo/dedicated server builds and the like.
